File: src/main/java/ru/sicampus/bootcamp2026/controller/ProfileController.java
package ru.sicampus.bootcamp2026.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;
import ru.sicampus.bootcamp2026.dto.request.ResetPasswordRequest;
import ru.sicampus.bootcamp2026.dto.request.UpdateAvatarRequest;
import ru.sicampus.bootcamp2026.dto.request.UserProfileRequest;
import ru.sicampus.bootcamp2026.dto.response.UserProfileResponse;
import ru.sicampus.bootcamp2026.model.CustomUserDetails;
import ru.sicampus.bootcamp2026.service.ProfileService;

@RestController
@RequestMapping("/api/v1/profile")
@RequiredArgsConstructor
public class ProfileController {

    private final ProfileService profileService;

    /**
     * Получение профиля текущего пользователя
     */
    @GetMapping
    public ResponseEntity<UserProfileResponse> getProfile(
            @AuthenticationPrincipal CustomUserDetails currentUser
    ) {
        UserProfileResponse profile = profileService.getProfile(currentUser.user().getId());
        return ResponseEntity.ok(profile);
    }

    /**
     * Обновление профиля текущего пользователя
     */
    @PutMapping
    public ResponseEntity<UserProfileResponse> updateProfile(
            @Valid @RequestBody UserProfileRequest request,
            @AuthenticationPrincipal CustomUserDetails currentUser
    ) {
        UserProfileResponse updatedProfile = profileService.updateProfile(
                currentUser.user().getId(),
                request
        );
        return ResponseEntity.ok(updatedProfile);
    }

    /**
     * Сброс пароля
     */
    @PutMapping("/reset-password")
    public ResponseEntity<Void> resetPassword(
            @Valid @RequestBody ResetPasswordRequest request,
            @AuthenticationPrincipal CustomUserDetails currentUser
    ) {
        /*
            TODO: Реализовать сброс пароля
        */
        throw new UnsupportedOperationException("Метод сброса пароля не реализован");
    }

    /**
     * Обновление аватара
     */
    @PutMapping("/avatar")
    public ResponseEntity<UserProfileResponse> updateAvatar(
            @Valid @RequestBody UpdateAvatarRequest request,
            @AuthenticationPrincipal CustomUserDetails currentUser
    ) {
        UserProfileResponse updatedProfile = profileService.updateAvatar(
                currentUser.user().getId(),
                request
        );
        return ResponseEntity.ok(updatedProfile);
    }

    /**
     * Получение списка всех пользователей для выбора участников встречи (поиск)
     */
    @GetMapping("/public/all")
    public ResponseEntity<Page<UserProfileResponse>> getAllUsers(
            @AuthenticationPrincipal CustomUserDetails currentUser,
            Pageable pageable
    ) {
        Page<UserProfileResponse> users = profileService.getAllUsers(currentUser.user().getId(), pageable);
        return ResponseEntity.ok(users);
    }

}

File: src/main/kotlin/ru/sicampus/bootcamp2026/controller/ProfileController.kt
package ru.sicampus.bootcamp2026.controller

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import ru.sicampus.bootcamp2026.dto.ProfileCreateDto
import ru.sicampus.bootcamp2026.dto.ProfileResponseDto
import ru.sicampus.bootcamp2026.dto.ProfileUpdateDto
import ru.sicampus.bootcamp2026.service.ProfileService

@RestController
@RequestMapping("/api/profiles")
class ProfileController(
    private val profileService: ProfileService
) {
    @GetMapping
    fun getAllProfiles(): ResponseEntity<List<ProfileResponseDto>> {
        return ResponseEntity.ok(profileService.getAllProfiles())
    }

    @GetMapping("/{id}")
    fun getProfileById(@PathVariable id: Long): ResponseEntity<ProfileResponseDto> {
        return ResponseEntity.ok(profileService.getProfileById(id))
    }

    @GetMapping("/user/{userId}")
    fun getProfileByUserId(@PathVariable userId: Long): ResponseEntity<ProfileResponseDto> {
        return ResponseEntity.ok(profileService.getProfileByUserId(userId))
    }

    @PostMapping
    fun createProfile(@RequestBody dto: ProfileCreateDto): ResponseEntity<ProfileResponseDto> {
        val createdProfile = profileService.createProfile(dto)
        return ResponseEntity.status(HttpStatus.CREATED).body(createdProfile)
    }

    @PutMapping("/{id}")
    fun updateProfile(
        @PathVariable id: Long,
        @RequestBody dto: ProfileUpdateDto
    ): ResponseEntity<ProfileResponseDto> {
        return ResponseEntity.ok(profileService.updateProfile(id, dto))
    }

    @DeleteMapping("/{id}")
    fun deleteProfile(@PathVariable id: Long): ResponseEntity<Void> {
        profileService.deleteProfile(id)
        return ResponseEntity.noContent().build()
    }
}
